modin/pandas/test/dataframe/test_indexing.py

```diff
@@ -562,6 +562,55 @@ def test_loc_multi_index():
     df_equals(modin_df.loc[modin_df.index[:7]], pandas_df.loc[pandas_df.index[:7]])
 
 
+# Better name suggestions?
+def test_loc_4358():
+    arrays = [
+        ["bar", "bar", "baz", "baz", "foo", "foo", "qux", "qux"],
+        ["one", "two", "one", "two", "one", "two", "one", "two"],
+    ]
+    columns = pd.MultiIndex.from_tuples(zip(*arrays), names=["a", "b"])
+    data = [np.random.randint(10, 50, len(columns)) for _ in range(30)]
+    modin_df = pd.DataFrame(data, columns=columns)
+    pandas_df = pandas.DataFrame(data, columns=columns)
+    df_equals(modin_df.loc[:, ("bar", "two")], pandas_df.loc[:, ("bar", "two")])
+
+
+def test_loc_multi_index_duplicate_keys():
+    modin_df = pd.DataFrame([1, 2], index=[["a", "a"], ["b", "b"]])
+    pandas_df = pd.DataFrame([1, 2], index=[["a", "a"], ["b", "b"]])
+    df_equals(modin_df.loc[("a", "b"), 0], pandas_df.loc[("a", "b"), 0])
+    df_equals(modin_df.loc[("a", "b"), :], pandas_df.loc[("a", "b"), :])
+
+
+def test_loc_multi_index_one_level():
+    modin_df = pd.DataFrame([1, 2], index=[["a", "b"], ["c", "c"]])
+    pandas_df = pd.DataFrame([1, 2], index=[["a", "b"], ["c", "c"]])
+    df_equals(modin_df.loc["a"], pandas_df.loc["a"])
+
+
+def test_loc_multi_index_both_axes():
+    multi_index = pd.MultiIndex.from_tuples(
+        [("r0", "rA"), ("r1", "rB")], names=["Courses", "Fee"]
+    )
+    cols = pd.MultiIndex.from_tuples(
+        [
+            ("Gasoline", "Toyota"),
+            ("Gasoline", "Ford"),
+            ("Electric", "Tesla"),
+            ("Electric", "Nio"),
+        ]
+    )
+    data = [[100, 300, 900, 400], [200, 500, 300, 600]]
+    df = pd.DataFrame(data, columns=cols, index=multi_index)
+    pdf = pandas.DataFrame(data, columns=cols, index=multi_index)
+    df_equals(df.loc[("r0", "rA"), :], pdf.loc[("r0", "rA"), :])
+    df_equals(df.loc[("r0"), :], pdf.loc[("r0"), :])
+    df_equals(df.loc[:, ("Gasoline")], pdf.loc[:, ("Gasoline")])
+    df_equals(df.loc[:, ("Gasoline", "Toyota")], pdf.loc[:, ("Gasoline", "Toyota")])
+    # Test below fails
+    # df_equals(df.loc[("r0"), ("Gasoline", "Toyota")], pdf.loc[("r0"), ("Gasoline", "Toyota")])
```

Comment:
What error appears here?

Reply:
There is a mismatch in the resulting DataFrames. Modin returns a single value and pandas returns a Series I believe. It seems to be an underlying issue with getting the value from `loc` with a DataFrame that has two MultiIndex columns. It didn't seem to be related to which axis we squeezed the result with.